### PR TITLE
[Fixes #36] ensure per user directories

### LINKED ISSUE #36: if-you-need-to-delete-this-open-an-issue-async-disk-cache
I have encountered this permission denied error fairly often while using ember cli on a shared environment with other ember developers. What other information do you need from me?

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,6 +21,10 @@ if (!heimdall.hasMonitor('async-disk-cache')) {
   heimdall.registerMonitor('async-disk-cache', function AsyncDiskCacheSchema() {});
 }
 
+var homedir = require('os').homedir().split(path.sep);
+var username = homedir[homedir.length - 1];
```
Comment (stefanpenner) [inline index.js]: this is for cross platform, but should i use `process.env.USER` ?

Comment (trentmwillis) [inline index.js]: Is `process.env.USER` guaranteed to be set? If so, seems good. Otherwise, maybe use it and fallback to the path splitting?

Comment (stefanpenner) [inline index.js]: in new node, we have a nice `os.userInfo().username`... would be nice if we had that everywhere. #someday

Comment (stefanpenner) [inline index.js]: well, fuckit good guy @sindresorhus  making things nice for us https://www.npmjs.com/package/username

**Pull request merged**